src/gleam/dynamic.gleam

```diff
@@ -153,6 +153,58 @@ pub external fn thunk(from: Dynamic) -> Result(fn() -> Dynamic, String) =
 pub external fn list(from: Dynamic) -> Result(List(Dynamic), String) =
   "gleam_stdlib" "decode_list"
 
+/// Check to see whether a Dynamic value is a result, and return the result if
+/// it is
+///
+/// ## Examples
+///
+///    > result(from(Ok(1)))
+///    Ok(Ok(from(1)))
+///
+///    > result(from(123))
+///    Error("Expected a result, got an int")
+///
+pub external fn result(
+  from: Dynamic,
+) -> Result(Result(Dynamic, Dynamic), String) =
+  "gleam_stdlib" "decode_result"
```

Comment:
Could we implement this in Gleam rather than Erlang?

Reply:
Done in https://github.com/gleam-lang/stdlib/pull/126/commits/3db2cbfd977c6bf1ff21e77221274476d137ac96